sys/include/net/gnrc/netif.h

```diff
@@ -252,6 +252,164 @@ gnrc_netif_t *gnrc_netif_iter(const gnrc_netif_t *prev);
  */
 gnrc_netif_t *gnrc_netif_get_by_pid(kernel_pid_t pid);
 
+/**
+ * @brief   Gets the (unicast on anycast) IPv6 addresss of an interface (if IPv6
+ *          is supported)
+ *
+ * @pre `netif != NULL`
+ * @pre `addrs != NULL`
+ * @pre `max_len >= sizeof(ipv6_addr_t)`
+ *
+ * @param[in] netif     The interface. May not be `NULL`.
```

Comment:
Judging from the assertions, shouldn't this be "Must not be `NULL`?

Reply:
Mhhh... In the past I always used "May"... Maybe I'm wrong, but AFAIK "must not" and "may not" are synonymous (meaning "you can not do that"; at least that is what my English-German dictionary is telling me), while their opposite "may" (it is optional to do that) and "must" (it is required to do that) are not. Maybe "may not" is a little bit weaker in the ears of a native speaker, but for consistency I would prefer to keep it this way or to change my older wordings first.

Reply:
Just realized however, that I used "Must not" [here as well](https://github.com/RIOT-OS/RIOT/pull/8253/files#diff-36d38c9fc81553f08febd7c298fd7168R350) ^^"

Reply:
Grammatically speaking I'd say you're completely correct here (not going to burn my fingers on English grammar). The reason I stumbled over this is because I'm interpreting these musts and mays as [rfc2119](https://tools.ietf.org/html/rfc2119) :)

I think together with the documented preconditions "may" is also clear enough.

Reply:
Mhhh, but RFC2119 only documents "Must", "Must not" and "May"... or do you mean *because* "may not" it isn't documented the precondition is unclear?





Sidenote: I'm not aware if the terminology of our doc MUST be according to RFC 2119 ;-).

Reply:
> Mhhh, but RFC2119 only documents "Must", "Must not" and "May"... or do you mean because "may not" it isn't documented the precondition is unclear?

To be honest, I never noticed that "May not" was missing. No, I rather think that with the documented `@pre`s, it is quite clear that the `netif` argument is mandatory.

> Sidenote: I'm not aware if the terminology of our doc MUST be according to RFC 2119 ;-).

Never said it should be according to that, only that I'm interpreting these things by default as RFC 2119. Whether that is correct is a whole different discussion. ;-)

Reply:
True, that's why I repeat myself:

> […] for consistency I would prefer to keep it this way or to change my older wordings first.

(I prefer to do it in a separate PR afterwards)